core/store/src/flat/store_helper.rs

```diff
@@ -183,11 +184,17 @@ pub fn iter_flat_state_entries<'a>(
     };
     store.iter_range(DBCol::FlatState, Some(&db_key_from), Some(&db_key_to)).filter_map(
         move |result| {
-            if let Ok((key, value)) = result {
-                let (_, trie_key) = decode_flat_state_db_key(&key).unwrap();
-                return Some((trie_key, value));
-            }
-            return None;
+            result
+                .map(|(key, value)| {
+                    (
+                        decode_flat_state_db_key(&key)
```

Comment:
Let's also wrap these `expect`s to `StorageError`s and make this function return `Result<impl Iterator<Item = (Vec<u8>, Box<[u8]>)> + 'a, StorageError>`.